src/shims/posix/fs.rs

```diff
@@ -108,8 +105,12 @@ impl FileDescriptor for io::Stdin {
         throw_unsup_format!("cannot seek on stdin");
     }
 
-    fn dup<'tcx>(&mut self) -> InterpResult<'tcx, io::Result<Box<dyn FileDescriptor>>> {
-        Ok(Ok(Box::new(io::stdin())))
+    fn close<'tcx>(self: Box<Self>, _communicate_allowed: bool) -> InterpResult<'tcx, io::Result<i32>> {
+        throw_unsup_format!("stdin cannot be closed(not supported by Rust)");
```

Comment:
```suggestion
        throw_unsup_format!("stdin cannot be closed");
```
We don't usually explain why something is not supported, and anyway that is not terribly helpful to the user I think.

(same for the other two)